common/persistence/sql/sqlTaskManager.go

```diff
@@ -255,9 +255,26 @@ func (m *sqlTaskManager) ListTaskQueue(request *persistence.ListTaskQueueRequest
 	var err error
 	var rows []sqlplugin.TaskQueuesRow
 	namespaceID := primitives.MustParseUUID(pageToken.NamespaceID)
-	for pageToken.ShardID < m.nShards {
+	var shardGreaterThan uint32
+	var shardLessThan uint32
+	for i := 0; i < m.taskScanPartitions; i++ {
```

Comment:
Can you refactor this logic into a separate component which returns ranges given partitionID and totalNumberOfPartitions.